### PR TITLE
add wechat_qrcode and unit test

### PR DESCRIPTION
**Pull request by lwqwag**
add wechat_qrcode and unit test

### CODE REVIEW

File: src/OpenCvSharpExtern/wechat_qrcode.h [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+CVAPI(ExceptionStatus) wechat_qrcode_Ptr_Get(cv::Ptr<cv::wechat_qrcode::WeChatQRCode>* obj, cv::wechat_qrcode::WeChatQRCode **returnValue)
+{
+    BEGIN_WRAP
+        * returnValue = obj->get();
+    END_WRAP
+}
+
+CVAPI(ExceptionStatus) detectAndDecode(cv::wechat_qrcode::WeChatQRCode* obj, cv::_InputArray* inputImage, std::vector<cv::Mat>* points, std::vector<std::string>* texts)
```
Comment (shimat) [inline src/OpenCvSharpExtern/wechat_qrcode.h]:
```suggestion
CVAPI(ExceptionStatus) wechat_qrcode_WeChatQRCode_detectAndDecode(cv::wechat_qrcode::WeChatQRCode* obj, cv::_InputArray* inputImage, std::vector<cv::Mat>* points, std::vector<std::string>* texts)
```

File: src/OpenCvSharpExtern/wechat_qrcode.h [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    END_WRAP
+}
+CVAPI(ExceptionStatus) wechat_qrcode_Ptr_Delete(cv::Ptr<cv::wechat_qrcode::WeChatQRCode>* obj)
+{
+    BEGIN_WRAP
+        delete obj;
+    END_WRAP
+}
+
+CVAPI(ExceptionStatus) wechat_qrcode_Ptr_Get(cv::Ptr<cv::wechat_qrcode::WeChatQRCode>* obj, cv::wechat_qrcode::WeChatQRCode **returnValue)
```
Comment (shimat) [inline src/OpenCvSharpExtern/wechat_qrcode.h]:
```suggestion
CVAPI(ExceptionStatus) wechat_qrcode_Ptr_WeChatQRCode_get(cv::Ptr<cv::wechat_qrcode::WeChatQRCode>* obj, cv::wechat_qrcode::WeChatQRCode **returnValue)
```

File: test/OpenCvSharp.Tests/wechat_qrcode/Wechat_QrcodeTest.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        public void Wechat_QrcodeDecodeTest()
+        {
+            Assert.True(File.Exists(_wechat_QCODE_detector_prototxt_path), $"DetectorPrototxt '{_wechat_QCODE_detector_prototxt_path}' not found");
+            Assert.True(File.Exists(_wechat_QCODE_detector_caffe_model_path), $"DetectorcaffeModel '{_wechat_QCODE_detector_caffe_model_path}' not found");
+            Assert.True(File.Exists(_wechat_QCODE_super_resolution_prototxt_path), $"SuperResolutionprototxt '{_wechat_QCODE_super_resolution_prototxt_path}' not found");
+            Assert.True(File.Exists(_wechat_QCODE_super_resolution_caffe_model_path), $"SuperResolutionCaffe_model '{_wechat_QCODE_super_resolution_caffe_model_path}' not found");
+#pragma warning disable CA2000 // 丢失范围之前释放对象
+            var wechatQrcode= WechatQrcode.Create(_wechat_QCODE_detector_prototxt_path, _wechat_QCODE_detector_caffe_model_path,
+                                                        _wechat_QCODE_super_resolution_prototxt_path, _wechat_QCODE_super_resolution_caffe_model_path);
+            var src = Cv2.ImRead(@"_data/image/qr_multi.png", ImreadModes.Grayscale);
```
Comment (shimat) [inline test/OpenCvSharp.Tests/wechat_qrcode/Wechat_QrcodeTest.cs]: This warning should not be disabled.
```suggestion
            using var wechatQrcode= WechatQrcode.Create(_wechat_QCODE_detector_prototxt_path, _wechat_QCODE_detector_caffe_model_path,
                                                        _wechat_QCODE_super_resolution_prototxt_path, _wechat_QCODE_super_resolution_caffe_model_path);
            using var src = Cv2.ImRead(@"_data/image/qr_multi.png", ImreadModes.Grayscale);
```

**Review by shimat — COMMENTED** [review]
Thank you for your contribution :+1:

File: src/OpenCvSharpExtern/wechat_qrcode.h (hunk 1 above)
Comment (lwqwag) [reply src/OpenCvSharpExtern/wechat_qrcode.h]: Thanks  and I will rename it.

File: src/OpenCvSharpExtern/wechat_qrcode.h (hunk 2 above)
Comment (lwqwag) [reply src/OpenCvSharpExtern/wechat_qrcode.h]: Thanks  and I will rename it.

File: test/OpenCvSharp.Tests/wechat_qrcode/Wechat_QrcodeTest.cs (hunk 3 above)
Comment (lwqwag) [reply test/OpenCvSharp.Tests/wechat_qrcode/Wechat_QrcodeTest.cs]: > This warning should not be disabled.
Thanks and I will disabled this warning

**Comment by shimat** [comment]
I'm sorry for the late reply. Thank you!

**Pull request merged**